tiledb/api/src/array/mod.rs

```diff
@@ -380,6 +382,113 @@ impl Array {
         Ok(Some(datatype))
     }
 
+    pub fn vacuum<S>(
+        ctx: &Context,
+        array_uri: S,
+        config: &Config,
+    ) -> TileDBResult<()>
+    where
+        S: AsRef<str>,
+    {
+        let c_array_uri = cstring!(array_uri.as_ref());
+        ctx.capi_call(|ctx| unsafe {
+            ffi::tiledb_array_vacuum(ctx, c_array_uri.as_ptr(), config.capi())
+        })?;
+        Ok(())
+    }
+
+    pub fn upgrade_version<S>(
+        ctx: &Context,
+        array_uri: S,
+        config: &Config,
+    ) -> TileDBResult<()>
+    where
+        S: AsRef<str>,
+    {
+        let c_array_uri = cstring!(array_uri.as_ref());
+        ctx.capi_call(|ctx| unsafe {
+            ffi::tiledb_array_upgrade_version(
+                ctx,
+                c_array_uri.as_ptr(),
+                config.capi(),
+            )
+        })?;
+        Ok(())
+    }
+
+    pub fn consolidate<S>(
+        ctx: &Context,
+        array_uri: S,
+        config: &Config,
+    ) -> TileDBResult<()>
+    where
+        S: AsRef<str>,
+    {
+        let c_array_uri = cstring!(array_uri.as_ref());
+        ctx.capi_call(|ctx| unsafe {
+            ffi::tiledb_array_consolidate(
+                ctx,
+                c_array_uri.as_ptr(),
+                config.capi(),
+            )
+        })?;
+        Ok(())
+    }
+
+    pub fn consolidate_fragments<S>(
+        ctx: &Context,
+        array_uri: S,
+        fragment_names: &[S],
+        config: &Config,
+    ) -> TileDBResult<()>
+    where
+        S: AsRef<str>,
+    {
+        let c_array_uri = cstring!(array_uri.as_ref());
+
+        let v = fragment_names
+            .iter()
+            .map(|fragment_name| Ok(cstring!(fragment_name.as_ref()).as_ptr()))
+            .map(|v| v.unwrap())
```

Comment:
The issue here is that on line 459 below I need to obtain a pointer to the vector's storage, so I need to eventually unwrap the result of cstring it seems to me. How else should I do it?

Reply:
If you do `collect::<TileDBResult<Vec<_>>>()?` with the question mark operator, then it will return `Err` if the result is an error and otherwise the expression evaluates to `Vec<_>`